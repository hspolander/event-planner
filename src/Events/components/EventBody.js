import React from 'react';
import { Paper, makeStyles } from '@material-ui/core';

import EventList from './EventList';

const useStyles = makeStyles((theme) => ({
  root: {},
  eventListPaper: {
    margin: 'auto',
    width: 600,
    height: 700,
  },
}));

const EventBody = () => {
  const classes = useStyles();
  return (
    <div className={classes.root}>
      <Paper elevation={2} className={classes.eventListPaper}>
        <EventList />
      </Paper>
    </div>
  );
};

export default EventBody;
