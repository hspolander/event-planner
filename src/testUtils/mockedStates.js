export const emptystate = {
  eventReducer: { events: null, fetching: false, error: false },
};

export const stateWithEvents = {
  eventReducer: {
    fetching: false,
    error: false,
    events: [
      {
        id: 3,
        startDate: '2019-02-26 10:00',
        endDate: '2019-02-26 12:00',
        location: 'Lagerhuset',
        activity: 'meeting',
        date: '2019-02-26 10:00:00',
      },
      {
        id: 1,
        startDate: '2019-02-26 12:00',
        endDate: '2019-02-26 13:00',
        location: 'Lagerhuset',
        activity: 'lunch',
        date: '2019-02-26 12:00:00',
      },
      {
        id: 2,
        startDate: '2019-02-26 22:00',
        endDate: '2019-02-28 07:00',
        location: 'online',
        activity: 'meeting',
        date: '2019-02-26 22:00:00',
      },
      {
        id: 2,
        startDate: '2019-02-26 22:00',
        endDate: '2019-02-28 07:00',
        location: 'online',
        activity: 'meeting',
        date: '2019-02-27 00:00:01',
      },
      {
        id: 2,
        startDate: '2019-02-26 22:00',
        endDate: '2019-02-28 07:00',
        location: 'online',
        activity: 'meeting',
        date: '2019-02-28 00:00:01',
      },
    ],
  },
};

export const stateWithError = {
  eventReducer: { events: null, fetching: false, error: true },
};

export const stateWhileLoading = {
  eventReducer: { events: null, fetching: true, error: false },
};
