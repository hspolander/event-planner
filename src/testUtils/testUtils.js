import React from 'react';
import { render } from '@testing-library/react';
import thunk from 'redux-thunk';
import { createStore, applyMiddleware } from 'redux';
import { Provider } from 'react-redux';
import rootReducer from '../rootReducer';

export default function renderWithRedux(
  ui,
  { initialState, store = createStore(rootReducer, initialState, applyMiddleware(thunk)) } = {}
) {
  return {
    ...render(<Provider store={store}>{ui}</Provider>),
    store,
  };
}

export * from '@testing-library/react';
