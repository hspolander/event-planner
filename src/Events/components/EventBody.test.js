import React from 'react';
import renderWithRedux, { cleanup } from '../../testUtils/testUtils';
import { mockedEmptyState } from '../../testUtils/mockedStates';

import EventBody from './EventBody';

describe('DashboardDetails', () => {
  afterEach(() => {
    cleanup();
  });

  it('should match snapshot', () => {
    const { container } = renderWithRedux(<EventBody />, {
      mockedEmptyState,
    });
    expect(container).toMatchSnapshot();
  });
});
