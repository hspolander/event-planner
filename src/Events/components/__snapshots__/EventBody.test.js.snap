// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardDetails should match snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root makeStyles-eventListPaper-1 MuiPaper-elevation2 MuiPaper-rounded"
  >
    <ul
      aria-labelledby="events-subheader"
      class="MuiList-root MuiList-padding MuiList-subheader"
      role="list"
    >
      <div
        class="MuiListSubheader-root MuiListSubheader-sticky MuiListSubheader-gutters"
        id="events-subheader"
      >
        Events
      </div>
    </ul>
  </div>
</div>
`;
